### PR TITLE
task/JM-7389: As a jury or bureau officer I want to generate a deferred list By Court (BE)

### PR DESCRIPTION
**Pull request by zacmnolan**
### JIRA link (if applicable) ###
https://centralgovernmentcgi.atlassian.net/browse/JM-6580

### Change description ###

BE implementation for Deferred list by court report

Added bug fix for unpaid attendance summary report not ordering correctly

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/juror/api/moj/report/grouped/UnpaidAttendanceSummaryReport.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ protected void preProcessQuery(JPAQuery<Tuple> query, StandardReportRequest requ
         query.where(QAppearance.appearance.locCode.eq(SecurityUtil.getLocCode()));
         query.where(QAppearance.appearance.attendanceDate.between(request.getFromDate(), request.getToDate()));
 
-        query.orderBy(QAppearance.appearance.poolNumber.asc(), QJuror.juror.jurorNumber.asc());
+        query.orderBy(QAppearance.appearance.attendanceDate.asc(), QJuror.juror.jurorNumber.asc());
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/report/grouped/UnpaidAttendanceSummaryReport.java:65]: Why has this report been updated?

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            .assertEquals(createResponse(0, new GroupedTableData()));
+    }
+
+    @Test
+    void positiveTypical() {
+        testBuilder()
+            .triggerValid()
+            .responseConsumer(this::verifyAndRemoveReportCreated)
+            .assertEquals(getTypicalResponse());
+    }
```
Comment (Ben-Edwards-cgi) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java:58]: Need a bureau flow (should see multiple courts)

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                .add("CHESTER (415)", List.of(
+                    new ReportLinkedMap<String, Object>()
+                        .add("deferred_to", "2023-01-06")
+                        .add("number_deferred", 1),
+                    new ReportLinkedMap<String, Object>()
+                        .add("deferred_to", "2023-01-07")
+                        .add("number_deferred", 2),
+                    new ReportLinkedMap<String, Object>()
+                        .add("deferred_to", "2023-01-08")
+                        .add("number_deferred", 1))));
```
Comment (Ben-Edwards-cgi) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java:72]: As chester has 2 satellite courts I would want to see some data in here for the satellite courts too

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java (hunk 2 above)
Comment (zacmnolan) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java:58]: Added

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java (hunk 3 above)
Comment (zacmnolan) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/grouped/DeferredListByCourtReportITest.java]: Added satellite court

**Review by Ben-Edwards-cgi — APPROVED** [review]

**Pull request merged**